scipy/io/idl.py

```diff
@@ -31,7 +31,7 @@
 
 import struct
 import numpy as np
-from numpy.compat import asstr
```

Comment:
You can't just remove this, `asstr` is used in the rest of the code here so should be adapted.

Please try to run tests locally, that way you'll catch such things.